portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java

```diff
@@ -0,0 +1,169 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns a list of comments */
+@WebServlet("/list-comments")
+public class ListCommentsServlet extends HttpServlet {
+
+  // GET parameters
+  private String commentPicId;
+  private int fetchQuantity;
+  private boolean isOrderByDate;
+
+  @Override
+  public void init() {
+    fetchQuantity = 5;
+    isOrderByDate = true;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    handleGetParams(request);
+
+    // A pic id is needed to continue
+    if (commentPicId.isEmpty()) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
+
+    Query fetchComments = buildQuery();
+
+    Iterable<Entity> commentsRetrieved = retrieveComments(fetchComments, fetchQuantity);
+
+    List<Comment> comments = createCommentsList(commentsRetrieved);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(convertListToJson(comments));
+  }
+
+  /**
+   * Return the request parameter or the default value if the parameter
+   * was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null || value.isEmpty()) {
+      return defaultValue;
+    }
+    return value;
+  }
+
+  /**
+   * Get and assign parameters of a request
+   */
+  private void handleGetParams(HttpServletRequest request) {
+    String imageId = getParameter(request, "imageId", "");
+    if (imageId.isEmpty()) {
+      commentPicId = "";
+      return; // Return here since getting other parameters will be worthless
+    } else {
+      commentPicId = imageId;
+    }
+
```

Comment:
```suggestion
    }
    commentPicId = imageId;
```
Since you return above, you can delete this else statement.

Reply:
Ahh, of course, I missed that one. Thanks!